server/mscalendar/settings_daily_summary.go

```
@@ -0,0 +1,259 @@
+	_, ok := value.(string)
+	if !ok {
+		return errors.New("trying to set Daily Summary Setting without a string value")
+	}
+	err := s.store.SetSetting(userID, s.id, value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *dailySummarySetting) Get(userID string) (interface{}, error) {
+	value, err := s.store.GetSetting(userID, s.id)
+	if err != nil {
+		return nil, err
+	}
+
+	_, ok := value.(*store.DailySummaryUserSettings)
+	if !ok {
+		return nil, errors.New("current value is not a Daily Summary Setting")
+	}
+
+	return value, nil
+}
+
+func (s *dailySummarySetting) GetID() string {
+	return s.id
+}
+
+func (s *dailySummarySetting) GetTitle() string {
+	return s.title
+}
+
+func (s *dailySummarySetting) GetDescription() string {
+	return s.description
+}
+
+func (s *dailySummarySetting) GetDependency() string {
+	return s.dependsOn
+}
+
+func (s *dailySummarySetting) GetSlackAttachments(userID, settingHandler string, disabled bool) (*model.SlackAttachment, error) {
+	title := fmt.Sprintf("Setting: %s", s.title)
+	currentValueMessage := "Disabled"
+
+	actions := []*model.PostAction{}
+	if !disabled {
+		dsumRaw, err := s.Get(userID)
+		if err != nil {
+			return nil, err
+		}
+		dsum := dsumRaw.(*store.DailySummaryUserSettings)
+
+		currentH := "8"
+		currentM := "00"
+		currentAPM := "AM"
+		fullTime := "8:00AM"
+		currentEnable := false
+		currentTextValue := "Not set."
+
+		if dsum != nil {
+			fullTime = dsum.PostTime
+			currentEnable = dsum.Enable
+			splitted := strings.Split(fullTime, ":")
+			currentH = splitted[0]
+			currentM = splitted[1][:2]
+			currentAPM = splitted[1][2:]
+			enableText := "Disabled"
+			if currentEnable {
+				enableText = "Enabled"
+			}
+			currentTextValue = fmt.Sprintf("%s (%s) (%s)", dsum.PostTime, dsum.Timezone, enableText)
+		}
+
+		timezone, err := s.getTimezone(userID)
+		if err != nil {
+			if dsum != nil {
+				timezone = dsum.Timezone
+			} else {
+				timezone = "UTC"
```

I'm not sure what benefit the user gets in setting their settings with the wrong timezone. 1/5 Instead of falling back on UTC, return an error here saying that we couldn't get their timezone from Microsoft.